tests/widgetresize.js

```diff
@@ -0,0 +1,324 @@
+/**
+ * @license Copyright (c) 2003-2020, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/* globals document, Event */
+
+import WidgetResize from '../src/widgetresize';
+
+// ClassicTestEditor can't be used, as it doesn't handle the focus, which is needed to test resizer visual cues.
+import ClassicEditor from '@ckeditor/ckeditor5-editor-classic/src/classiceditor';
+import ArticlePluginSet from '@ckeditor/ckeditor5-core/tests/_utils/articlepluginset';
+
+import {
+	toWidget
+} from '../src/utils';
+import {
+	setData as setModelData
+} from '@ckeditor/ckeditor5-engine/src/dev-utils/model';
+
+import Rect from '@ckeditor/ckeditor5-utils/src/dom/rect';
+
+describe( 'WidgetResize', () => {
+	let editor, editorElement, widget, mouseListenerSpies;
+
+	const commitStub = sinon.stub();
```

Comment:
I don't like that the initialization code is split like that. Some parts are directly in describe(), some are in before(), some in beforeEach(). I started analysing why and whether it's correct but then I understood that if it was all in beforeEach() as we normally do, I wouldn't be thinking about this at all. So, I prefer consistency.